### DISCUSSION TITLE
Code Project v 2.8.0 native on Windows 10 service doesn't start

### DISCUSSION THREAD

**Discussion by sw6661** [codeproject/CodeProject.AI-Server#148] [Q&A] [answered]
Suddenly, cannot start the CodeProject AI server service.
I have tried disabling anti virus, rebooting several times, uninstalling and reinstalling ver 2.8.0, downloaded 2.8.0 from this GitHub website with no good results.
The only error is 1503, the service did not start in a timely manner.
I have been able to get the 2.8.0 Docker version running under WSL2, but I am trying to remember how to save the settings within the docker VE.
Very busy today, will continue later.
Any ideas would be appreciated.
(I think I had this before, unfortunately, the original Code Project Server discussion page is gone.)

**Comment by N6ZZY** [comment]
I had a similar issue after upgrading my two systems running 2.6.5 to 2.8.  I needed to download windows .net 8 SDK and install.  Once I did that the service started fine.  Open windows event viewer and look at the windows logs (Application), it should show error messages why the service failed to run.

**Reply by sw6661** [reply]
I had been running 2.8.0 for quite a while, I had installed .Net verify 8 to get it to run. Aside from an insurance a couple of weeks ago where the AI-Server service was using almost all the memory in the machine, it was fine. I've got it working now with docker in WSL2 and it's ok. I'm curious. I thought it was something to do with the browser. I had this about a year ago. I wish they hadn't shut down the original web page. Lot's of good info there.

**Reply by sw6661** [reply]
Boy, I hate auto correct.
"instance" not insurance.

**Reply by sw6661** [reply]
Nothing at all in the application log related to this.
It seems that on install, there is a folder and files missing.
.Net is up to date. Seems the modules folder is missing.

Setup produces this:


C:\Program Files\CodeProject\AI>setup

             Installing CodeProject.AI Analysis Module

======================================================================

                   CodeProject.AI Installer

======================================================================

276.1Gb of 487Gb available on no (Windows 10 x86_64 - windows)

General CodeProject.AI setup

Creating Directories...done

GPU support

CUDA Present...No
ROCm Present...No
Checking for .NET 8.0...Checking runtimes...All good. .NET is 8.0.1
Cannot find file "C:\Program Files\CodeProject\AI\modules\AI\modulesettings.json"

Reading  settings.......done
Installing module

This module cannot be installed on this system

Setup complete

Total setup time 00:00:02.86

C:\Program Files\CodeProject\AI>

**Reply by sw6661** [reply]
Continuing to try to properly install the native Windows CodeProject.AI Server application.
I have uninstalled the CPAI application several times.
It seems that when I uninstall the application, the CodeProject.AI Server service listing in the services.msc listing never leaves.
In order to attempt to do a complete install with the \Modules\Modulesettings.json file, I ran the earlier install that I had downloaded from the original CPAI website, before it was moved to GitHub.
I also selected a different folder to install the application.
On doing this, it seems that the SC command that is hard coded in the install for CPAI is C:\ProgramFiles\CodeProject|AI\Server.
So If you select a different folder to install the application in a different folder, the service will not run, but with a different error.
The service cannot find the file.
So I searched a little how to change the location of the application in the service setting.
Running SC QC produced the following:

C:\Windows\system32>sc qc CodeProject.AI Server
[SC] OpenService FAILED 1060:

The specified service does not exist as an installed service.

Running the SC command to list all services does in fact indicate that there is no CodeProject.AI Server service.

I am now trying to research how to remove the CodeProject.AI Server listing from the list of services shown in the services.msc list.

I would really hate to have to reinstall Windows on this machine in order to try to resolve this.

**Reply by sw6661** [reply]
Still trying. Uninstall and reinstall, the service does not exist.
Tried to manually install the service.
Resulted with this:

C:\Windows\Microsoft.NET\Framework\v4.0.30319>Installutil.exe "C:\Program Files\CodeProject\AI\Server\CodeProject.AI.Server.exe"
Microsoft (R) .NET Framework Installation utility Version 4.8.9037.0
Copyright (C) Microsoft Corporation.  All rights reserved.

Exception occurred while initializing the installation:
System.BadImageFormatException: Could not load file or assembly 'file:///C:\Program Files\CodeProject\AI\Server\CodeProject.AI.Server.exe' or one of its dependencies. **The module was expected to contain an assembly manifest..**

C:\Windows\Microsoft.NET\Framework\v4.0.30319>

**Reply by sw6661** [reply]
Just on a whim, I went way way back to version 2.5.1.
Install went fine, the old way, service installed and started as it should, running YOLOv5.net and returning responses to BlueIris as it should.
Ocasionally seeing this in the log:

2024-11-07 15:09:33: Object Detection (YOLOv5 .NET): Command completed in 268 ms. (command timing) in Object Detection (YOLOv5 .NET)
2024-11-07 15:09:48: ObjectDetectionYOLOv5Net.exe: Unable to get request from objectdetection_queue for ObjectDetectionYOLOv5Net
2024-11-07 15:09:48: ObjectDetectionYOLOv5Net.exe: Pausing on error for 30 secs.
2024-11-07 15:09:48: ObjectDetectionYOLOv5Net.exe: Unable to get request from objectdetection_queue for ObjectDetectionYOLOv5Net
2024-11-07 15:09:48: ObjectDetectionYOLOv5Net.exe: Pausing on error for 30 secs.

Also, I do not see the counter in the status bar incrementing, but also running on CPU.

I think this was a bug in that version.

At least it is running, the service is starting etc.

**Reply by sw6661** [reply]
Version 2.5.4 is working fine now. No problem with installing or starting the service. I'll stick with that for now.

**Comment by tbc725** [comment, marked as answer]
I was having similar problems with 2.8.0 that I cannot start the CPAI services. My situation is different than yours as mine 2.8.0 was working and I mistakenly tell VS Code to uninstall something on the ".NET" for me and it stopped working after that.

I checked the Events Viewer and found the errors were saying it need 8.0 version of .Net Core.  I downloaded and installed the "ASP.Net Core Runtime 8.0.10" and ".NET Runtime 8.0.11" from "https://dotnet.microsoft.com/en-us/download/dotnet/8.0" and was working fine after that.

**Reply by sw6661** [reply]
TBC725, thank you for the link. I ran the install, but after installing ver 2.8.0, I had to restart the machine, restart the service several times, and change Blue Iris back and forth to Default to Custom several times to get Blue Iris to properly talk to CPAI.
I hate to be a nag, but I still do not understand how when I installed version 2.8.0 on 8/15/2024, I did install .NET version 8 as required, and CPAI worked with no issues, service started on reboot with no problem etc, until approximately 10/30/2024. then it quit, with the service never to be seen again. I don't get it.
I do really miss Deepstack (on *nix). I remember hearing a lot of discussion at that time that "there is no development being done on Deepstack".
My experience tells me that sometimes that is a good thing. The code actually doesn't change itself. I have stories about using software that is years old, and also software that is constantly under development having mistakes "built in". I have 40 some years with different platforms, software, operating systems etc.
